scripts/hubspot-deploy/hubspot-publisher.js

```diff
@@ -28,60 +36,50 @@ class HubSpotPublisher {
   }
 
   async uploadPage(academyPage) {
-    return fs.readFile(academyPage.fileLocation, 'utf8').then(html => {
-      academyPage.setHtml(html);
-
-      if(academyPage.hubSpotId) {
-        return this.hubSpotApi.updatePage(
-          academyPage.hubSpotId,
-          {
-              title: academyPage.getTitle(),
-              headHtml: academyPage.getCSSLinks(),
-              bodyHtml: academyPage.getPageContents(),
-              metaDescription: academyPage.getMetaDescription()
-          }
-        );
-      }
-      else {
-        return this.hubSpotApi.createPage({
+    if(academyPage.hubSpotId) {
+      return this.hubSpotApi.updatePage(
+        academyPage.hubSpotId,
+        {
             title: academyPage.getTitle(),
             headHtml: academyPage.getCSSLinks(),
             bodyHtml: academyPage.getPageContents(),
-            slug: academyPage.slug,
             metaDescription: academyPage.getMetaDescription()
-        });
-      }
-    });
+        }
+      );
+    }
+    else {
+      return this.hubSpotApi.createPage({
+          title: academyPage.getTitle(),
+          headHtml: academyPage.getCSSLinks(),
+          bodyHtml: academyPage.getPageContents(),
+          slug: academyPage.slug,
+          metaDescription: academyPage.getMetaDescription()
+      });
+    }
   }
 
   async publish(){
     const pagesCurrentlyOnHubSpot = await this.hubSpotApi.getPages();
     const pagesForHubSpotUpload = (await this.getPagesToUpload());
-    const pagesToBeDeleted = pagesCurrentlyOnHubSpot.filter(pageCurrentlyOnHubSpot =>
-      !pagesForHubSpotUpload.find(pageForHubSpotUpload => pageCurrentlyOnHubSpot.slug === pageForHubSpotUpload.slug)
-    );
 
-    pagesForHubSpotUpload.forEach(page => {
+    await Promise.all(pagesForHubSpotUpload.map(page => {
       const localPageOnHubSpot = pagesCurrentlyOnHubSpot.find(pageCurrentlyOnHubSpot =>
         pageCurrentlyOnHubSpot.slug === page.slug
       );
+
       if(localPageOnHubSpot){
         page.hubSpotId = localPageOnHubSpot.id;
       }
-      this.uploadPage(page)
-    });
-    promptDeleteFiles(pagesToBeDeleted,
-      () => {
-        // delete all
-        pagesToBeDeleted.forEach(pageToBeDeleted => this.hubSpotApi.deletePage(pageToBeDeleted.id))
-      },
-      async () => {
-        // choose which to delete
-        for(let pageToBeDeleted of pagesToBeDeleted){
-          await confirmDeleteFile(pageToBeDeleted.slug, () => this.hubSpotApi.deletePage(pageToBeDeleted.id))
-        }
-      }
-    )
+
+      return this.uploadPage(page)
+    }));
+
+    const pagesToBeDeleted = pagesCurrentlyOnHubSpot.filter(pageCurrentlyOnHubSpot =>
+      !pagesForHubSpotUpload.find(pageForHubSpotUpload => pageCurrentlyOnHubSpot.slug === pageForHubSpotUpload.slug)
+    );
+
+    // TODO: Remove extra files from the /academy directory
```

Comment:
is this TODO okay to be merged?

Reply:
yes. That's a far-future todo. In theory, this will leave orphaned files in hubspot if they are removed from academy. That's not really a problem, but would be nice to fix at some point.